src/JBCCIPSucker.sol

```diff
@@ -0,0 +1,196 @@
+    function _buildCCIPMessage(
+        address _receiver,
+        JBMessageRoot memory _root,
+        address _token,
+        uint256 _amount,
+        address _feeTokenAddress
+    ) private pure returns (Client.EVM2AnyMessage memory) {
+        // Set the token amounts
+        Client.EVMTokenAmount[] memory tokenAmounts = new Client.EVMTokenAmount[](1);
+        tokenAmounts[0] = Client.EVMTokenAmount({token: _token, amount: _amount});
+        // Create an EVM2AnyMessage struct in memory with necessary information for sending a cross-chain message
+        return Client.EVM2AnyMessage({
+            receiver: abi.encode(_receiver), // ABI-encoded receiver address
+            data: abi.encode(_root), // ABI-encoded string
+            tokenAmounts: tokenAmounts, // The amount and type of token being transferred
+            extraArgs: Client._argsToBytes(
+                // Additional arguments, setting gas limit
+                Client.EVMExtraArgsV1({gasLimit: 300_000})
+            ),
+            // Set the feeToken to a feeTokenAddress, indicating specific asset will be used for fees
+            feeToken: _feeTokenAddress
+        });
+    }
+
+    /// @notice The entrypoint for the CCIP router to call. This function should
+    /// never revert, all errors should be handled internally in this contract.
+    /// @param any2EvmMessage The message to process.
+    /// @dev Extremely important to ensure only router calls this.
+    function ccipReceive(Client.Any2EVMMessage calldata any2EvmMessage) external override onlyRouter {
+        _ccipReceive(any2EvmMessage);
+    }
+
+    function _ccipReceive(Client.Any2EVMMessage memory any2EvmMessage) internal override {
+        // Decode the message root from the peer
+        JBMessageRoot memory root = abi.decode(any2EvmMessage.data, (JBMessageRoot));
+        address origin = abi.decode(any2EvmMessage.sender, (address));
+
+        // Make sure that the message came from our peer.
+        if (origin != address(this)) revert NOT_PEER();
+
+        // Increase the outstanding amount to be added to the project's balance by the amount being received.
+        amountToAddToBalance[root.token] += root.amount;
+
+        // TODO: Is this necessary anymore?
+        // If the received tree's nonce is greater than the current inbox tree's nonce, update the inbox tree.
+        // We can't revert because this could be a native token transfer. If we reverted, we would lose the native tokens.
+        if (root.remoteRoot.nonce > inbox[root.token].nonce) {
+            inbox[root.token] = root.remoteRoot;
+            emit NewInboxTreeRoot(root.token, root.remoteRoot.nonce, root.remoteRoot.root);
+        }
+    }
+
+    /// @notice Checks if the `sender` (`msg.sender`) is a valid representative of the remote peer.
+    /// @param sender The message's sender.
+    function _isRemotePeer(address sender) internal view override onlyRouter returns (bool _valid) {
```

Comment:
There is a slight discrepancy here, between the comment above and the modifier vs the argument of the function and the other suckers their logic. 

The comment suggest it only checks the `msg.sender`, and the modifier does only check the `msg.sender.` But for the other suckers this function checks that the address passed is the remotePeer (or a representative of it).

Reply:
Hear you. Think we can just void the function since the only use is for ```fromRemote()``` which can't be used by CCIP receivers? They have to use ccipreceive.